climada_petals/hazard/river_flood.py

```diff
@@ -163,47 +162,40 @@ def from_nc(cls, dph_path=None, frc_path=None, origin=False,
                     iso_codes = u_coord.region2isos(reg)
                     # envelope containing counties
                     cntry_geom = u_coord.get_land_geometry(iso_codes)
-                    # TODO: replace this call to `set_raster` with a call to `from_raster` when
-                    # replacing `set_from_nc` with a @classmethod `from_nc`
-                    haz.set_raster(haz_type=HAZ_TYPE,
-                                    files_intensity=[dph_path],
-                                    files_fraction=[frc_path],
-                                    band=bands.tolist(),
-                                    geometry=cntry_geom)
+                    haz = cls.from_raster(files_intensity=[dph_path],
+                                          files_fraction=[frc_path],
+                                          band=bands.tolist(),
+                                          geometry=cntry_geom.geoms)
                     # self.centroids.set_meta_to_lat_lon()
                 else:
                     cntry_geom = u_coord.get_land_geometry(countries)
-                    # TODO: replace this call to `set_raster` with a call to `from_raster` when
-                    # replacing `set_from_nc` with a @classmethod `from_nc`
-                    haz.set_raster(haz_type=HAZ_TYPE,
-                                    files_intensity=[dph_path],
-                                    files_fraction=[frc_path],
-                                    band=bands.tolist(),
-                                    geometry=cntry_geom)
+                    haz = cls.from_raster(files_intensity=[dph_path],
+                                          files_fraction=[frc_path],
+                                          band=bands.tolist(),
+                                          geometry=cntry_geom.geoms)
                     # self.centroids.set_meta_to_lat_lon()
 
         elif shape:
             shapes = gpd.read_file(shape)
 
             rand_geom = shapes.geometry[0]
 
-            # TODO: replace this call to `set_raster` with a call to `from_raster` when
-            # replacing `set_from_nc` with a @classmethod `from_nc`
-            haz.set_raster(haz_type=HAZ_TYPE,
-                            files_intensity=[dph_path],
-                            files_fraction=[frc_path],
-                            band=bands.tolist(),
-                            geometry=rand_geom)
+            if isinstance(rand_geom, MultiPolygon):
+                rand_geom = rand_geom.geoms
+            elif isinstance(rand_geom, Polygon) or not isinstance(rand_geom, Iterable):
+                rand_geom = [rand_geom]
+
+            haz = cls.from_raster(files_intensity=[dph_path],
+                                  files_fraction=[frc_path],
+                                  band=bands.tolist(),
+                                  geometry=rand_geom)
             return
```

Comment:
```suggestion
```

Reply:
I'm not sure - continue with `haz` or return it right away?

Reply:
Well spotted! This is actually a bug from the conversion to the `from_*` functions. Furthermore, I think that returning here was also unintended in the original code because otherwise attributes like `units` are not properly set.